README.md

```diff
@@ -129,6 +137,87 @@ Similarly, to put a clojure data structure on the producer channel:
 (>!! producer-chan ["k2" "v2" "events"])
 ```
 
+## Consumer Decorator
+The consumer decorator allows running custom logic on the consumer polling thread.  
+This allows custom control on the consumer behavior including manual offset management.  
+Custom decorator logic may require different consumer configurations.  
+for example when managing the offset manually, auto-commit should usually set to false.    
+
+In this example we demonstare how to enable pause/resume of the consumer:
+```clojure
+(ns consumer-decorator-example
+  (:require [clojure.core.async :as async]
+            [ketu.async.source :as source]
+            [ketu.async.sink :as sink]))
+
+(let [commands-chan (async/chan 10)
+      consumer-chan (async/chan 10)
+      consumer-opts {:name                           "consumer-example"
+                     :brokers                        "broker1:9092"
+                     :topic                          "example"
+                     :group-id                       "example"
+                     :value-type                     :string
+                     :shape                          :value
+                     :ketu.source/consumer-decorator (fn [consumer-ctx poll-fn]
+                                                       (loop []
```

Comment:
I suggest adding a comment explaining that this decorator processes all immediately available commands in the `commands-chan`, and only then calls `(poll-fn)`.

Reply:
this is an implementation detail of a specific consumer decorator.
you can implement a decorator that doesn't have a command channel

Reply:
Since this example in the main one reader will see, I think we should some more context here